baseServer.go

```diff
@@ -29,7 +29,8 @@ type BaseServer[T IAgent[T]] struct {
 	//lock which stops race conditions when updating agentStoppedTalkingMap
 	agentMapRWMutex sync.RWMutex
 	//stops multiple sends to messagingFinished during a round
-	doneChannelOnce sync.Once
+	doneChannelOnce    sync.Once
+	asyncMessagingFlag bool
```

Comment:
this is already a flag. you dont need to name it as one. 

`shouldRunAsyncMessaging` is better